lib/palindrome_permutation.rb

```diff
@@ -1,4 +1,20 @@
 
 def palindrome_permutation?(string)
-  raise NotImplementedError, "palindrome_permutation? not implemented"
+  hash = {}
+  # keys are the letters and their values are their counts
+  string.each_char do |letter|
+    if hash[letter].nil?
+      hash[letter] = 1
+    else
+      hash[letter] += 1
+    end
+  end
+
+  if string.length.even? && hash.keys.length == string.length/2 
+    return true 
+  elsif string.length.odd? && hash.keys.length == string.length / 2 + 1
+    return true
+  else 
+    return false
+  end
```

Comment:
This might work better if you simply counted the number of odd values in the hash.